src/cloudant/_common_util.py

```diff
@@ -124,19 +125,22 @@
     'group_sort': (STRTYPE, list),
     'include_docs': bool,
     'limit': (int, NONETYPE),
-    'query': (STRTYPE, int),
-    'q': (STRTYPE, int),
+    'query': (STRTYPE, int, LONGTYPE),
```

Comment:
@smithsz It may be better to change LONGTYPE to  `LONGTYPE = long if PY2 else int` since all integers are long ints in Py3.